tests/Provider/ConcreteProviderTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Yiisoft\EventDispatcher\Tests\Provider;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\EventDispatcher\Provider\ConcreteProvider;
+use Yiisoft\EventDispatcher\Tests\Event\ClassInterface;
+use Yiisoft\EventDispatcher\Tests\Event\ClassItself;
+use Yiisoft\EventDispatcher\Tests\Event\Event;
+use Yiisoft\EventDispatcher\Tests\Event\ParentClass;
+use Yiisoft\EventDispatcher\Tests\Event\ParentInterface;
+
+use function array_slice;
+
+class ConcreteProviderTest extends TestCase
+{
+    /**
+     * @var ConcreteProvider
+     */
+    private ConcreteProvider $concreteProvider;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->concreteProvider = new ConcreteProvider();
+    }
+
+    public function testListenerForEventIsReturned(): void
+    {
+        $listener = fn () => null;
+
+        $this->concreteProvider->attach(Event::class, $listener);
+
+        $listeners = $this->concreteProvider->getListenersForEvent(new Event());
+
+        $this->assertContains($listener, $listeners);
```

Comment:
Please use `assertEquals`

Reply:
I don't get what you mean, IMHO `assertContains` does the job perfectly here.

Plus it often is considered bad practice to use `assertEquals` as it doesn't handle types in a strong way (`assertSame` does).

What would you do there instead?

Reply:
Need to check that `$listeners` contains only one listener.

`$this->assertEquals([$listener], $listeners);`

Also need to make one more test case that will check multiple listenters

Reply:
Ok, so I think adding `$this->assertCount(1, $listeners);` would make more sense then.

I'll also add a test case for multiple listeners.